### ISSUE TITLE
Latest update dependency breaking OSX

### ISSUE THREAD

**Issue by opoudjis** [fontist/ffi-libarchive-binary#2]
The latest release of this gem will not work on OSX without `brew install lz4`:

````
LoadError:
  Could not open library '/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.so.13.dylib': dlopen(/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.so.13.dylib, 5): image not found.
  Could not open library '/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.13.dylib': dlopen(/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.13.dylib, 5): image not found.
  Could not open library '/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive-13.dylib': dlopen(/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive-13.dylib, 5): image not found.
  Could not open library '/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.so.dylib': dlopen(/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.so.dylib, 5): image not found.
  Could not open library '/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.dylib': dlopen(/Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.dylib, 5): Library not loaded: /usr/local/opt/lz4/lib/liblz4.1.dylib
    Referenced from: /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary/libarchive.dylib
    Reason: image not found
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-1.15.0/lib/ffi/library.rb:145:in `block in ffi_lib'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-1.15.0/lib/ffi/library.rb:99:in `map'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-1.15.0/lib/ffi/library.rb:99:in `ffi_lib'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary.rb:19:in `ffi_lib'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-1.0.17/lib/ffi-libarchive/archive.rb:11:in `<module:C>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-1.0.17/lib/ffi-libarchive/archive.rb:4:in `<module:Archive>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-1.0.17/lib/ffi-libarchive/archive.rb:3:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-1.0.17/lib/ffi-libarchive.rb:56:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-1.0.17/lib/ffi-libarchive.rb:56:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary.rb:24:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary.rb:24:in `<module:LibarchiveBinary>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/ffi-libarchive-binary-0.2.1-x86_64-darwin/lib/ffi-libarchive-binary.rb:7:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/excavate-0.2.2/lib/excavate/extractors/xar_extractor.rb:1:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/excavate-0.2.2/lib/excavate/extractors/xar_extractor.rb:1:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/excavate-0.2.2/lib/excavate/extractors.rb:9:in `require_relative'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/excavate-0.2.2/lib/excavate/extractors.rb:9:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/excavate-0.2.2/lib/excavate.rb:4:in `require_relative'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/excavate-0.2.2/lib/excavate.rb:4:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/fontist-1.8.12/lib/fontist/font_installer.rb:2:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/fontist-1.8.12/lib/fontist/font_installer.rb:2:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/fontist-1.8.12/lib/fontist/font.rb:1:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/fontist-1.8.12/lib/fontist/font.rb:1:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/fontist-1.8.12/lib/fontist.rb:10:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/fontist-1.8.12/lib/fontist.rb:10:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/metanorma-1.2.8/lib/metanorma/compile.rb:6:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/metanorma-1.2.8/lib/metanorma/compile.rb:6:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/metanorma-1.2.8/lib/metanorma.rb:12:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/metanorma-1.2.8/lib/metanorma.rb:12:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/isodoc-1.5.5/lib/isodoc/pdf_convert.rb:4:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/isodoc-1.5.5/lib/isodoc/pdf_convert.rb:4:in `<top (required)>'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/isodoc-1.5.5/lib/isodoc.rb:21:in `require'
# /Users/nickn/.rvm/gems/ruby-2.7.2/gems/isodoc-1.5.5/lib/isodoc.rb:21:in `<top (required)>'
# ./lib/asciidoctor/standoc/base.rb:7:in `require'
# ./lib/asciidoctor/standoc/base.rb:7:in `<top (required)>'
# ./lib/asciidoctor/standoc/converter.rb:4:in `require'
# ./lib/asciidoctor/standoc/converter.rb:4:in `<top (required)>'
# ./lib/metanorma-standoc.rb:3:in `require_relative'
# ./lib/metanorma-standoc.rb:3:in `<top (required)>'
# ./spec/spec_helper.rb:20:in `require'
# ./spec/spec_helper.rb:20:in `<top (required)>'
# ./spec/metanorma/processor_spec.rb:1:in `require'
# ./spec/metanorma/processor_spec.rb:1:in `<top (required)>'
````

Once I install the lz4 library, I can proceed with using the metanorma stack.

Alert @CAMOBAP in case GitHub Actions and Metanorma build scripts need to be updated. Binary dependencies should be avoided if possible.

**Comment by CAMOBAP** [comment]
For users who install metanorma with `brew` this can be easily fixed by adding `lz4` dependency. Also, this doesn't an issue for GHA (looks like `lz4` preinstalled there).

A possible way to fix:
 - Add `brew install lz4` logic in https://github.com/fontist/ffi-libarchive-binary/blob/master/lib/ffi-libarchive-binary.rb#L10
 - Update recipe in this repo and statically link `lz4`
 - Build without `lz4` maybe another archive backend library can handle this (IDK it it will works for us or not) https://github.com/libarchive/libarchive/blob/master/configure.ac#L391


@ronaldtse do we have some users who use our toolchain as a ruby gem? If so I think, as last chance, we can add note to metanorma.com site

@alexeymorozov @ronaldtse how do you think about this?

**Comment by alexeymorozov** [comment]
I'll try to build it without `lz4`, and even try to reduce other dependencies to a minimum set.

**Comment by alexeymorozov** [comment]
Temporarily will release `excavate` without `ffi-libarchive-binary`, so it can be installed and used without pinning an old version.

**Comment by alexeymorozov** [comment]
Released ffi-libarchive-binary v0.2.2 and excavate v0.2.4 which contain much less dependencies than before.

List of remaining dependencies: https://github.com/fontist/ffi-libarchive-binary#dependencies
List of disabled libs: https://github.com/fontist/ffi-libarchive-binary/blob/master/lib/ffi-libarchive-binary/libarchive_recipe.rb#L25

**Comment by ronaldtse** [comment]
@CAMOBAP yes, I know at least one tool that is dependent on metanorma as a library.
